sybl/client/sybl_client.py

```diff
@@ -212,50 +215,64 @@ def load_model(self, email: str, model_name: str) -> None:
     def _process_job(self) -> None:
         logger.info("PROCCESSING JOB")
 
+        # Get message from message stack
         if self._message_stack:
             data: Dict = self._message_stack.pop()
 
+        # Make sure the dataset ia actually there
         assert "Dataset" in data
 
+        # Get training and prediction datasets
         train = data["Dataset"]["train"]
         predict = data["Dataset"]["predict"]
 
         train_pd = pd.read_csv(io.StringIO(train))
         predict_pd = pd.read_csv(io.StringIO(predict))
 
-        predict_rids = None
-
-        if "record_id" in train_pd.columns:
-            # Take record ids from training set
-            train_pd = train_pd.drop(["record_id"], axis=1)
-            logger.debug("Training Data: %s", train_pd.head())
-
-            # Take record ids from predict set and store for later
-            predict_rids = predict_pd[["record_id"]]
-            logger.debug("Predict Record IDs: %s", predict_rids.head())
-
-            predict_pd = predict_pd.drop(["record_id"], axis=1)
-            logger.debug("Predict Data: %s", predict_pd.head())
-        else:
-            raise AttributeError("Datasets must have record ids for each row")
+        # Prepare the datasets for callback
+        train_pd, predict_pd, predict_rids = self._prepare_datasets(
+            train_pd, predict_pd
+        )
 
         # Check the user has specified a callback here to satisfy mypy
         assert self.callback is not None
 
         predictions = self.callback(train_pd, predict_pd)
 
+        logger.debug("Predictions: %s", predictions.head())
+
         # Attatch record ids onto predictions
-        if predict_rids is not None:
-            predictions["record_id"] = predict_rids
-            cols = predictions.columns.tolist()
-            cols = cols[-1:] + cols[:-1]
-            predictions = predictions[cols]
+        predictions["record_id"] = predict_rids
+        cols = predictions.columns.tolist()
+        cols = cols[-1:] + cols[:-1]
```

Comment:
I looked online at a better way of writing this as it isnt completely clear what it does, I think this is clearer
```suggestion
        cols = cols.insert(0, cols.pop())
```

Reply:
Turns out this is incorrect and doesn't work

Reply:
```suggestion
        cols.insert(0, cols.pop())
```